### PR TITLE
fix: adjust return order on normalizeSelectAllField function

### PR DESCRIPTION
**Pull request by tomrndom**
ref: https://app.clickup.com/t/86b9ydmgt

Signed-off-by: Tomás Castillo <tcastilloboireau@gmail.com>

## Summary by CodeRabbit

* **Bug Fixes**
  * Select-all behavior now respects an explicit "all selected" flag even when items are empty, null, or undefined, avoiding incorrect deactivation and ensuring consistent selection state.

* **Tests**
  * Added parameterized tests covering empty/null/undefined item lists to validate the normalized select-all behavior.

[Review Change Stack](https://app.coderabbit.ai/change-stack/fntechgit/summit-admin/pull/948?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: src/utils/methods.js [hunk 1]
```diff
@@ -601,19 +601,19 @@ export const normalizeSelectAllField = (
   listName,
   allSelected
 ) => {
-  if (!items?.length)
+  const isAllSelected = allSelected ?? items?.includes("all");
```
Comment (smarcet) [inline src/utils/methods.js:604]: @tomrndom 
No test coverage for the fixed edge case
 The exact bug scenario (allSelected=true with empty/null/undefined items) has no test — the three changed paths can regress silently on any future refactor of this utility used across 7+ call sites .
Add tests for normalizeSelectAllField([], "apply_to_all", "items", true), same for null and undefined items — each should assert { apply_to_all: true, items: [] }

Comment (smarcet) [inline src/utils/methods.js:604]: @tomrndom 
Add a test for normalizeSelectAllField([], "flag", "list", false) asserting { flag: false, list: [] } to pin this behavior

**Review by smarcet — CHANGES_REQUESTED** [review]
@tomrndom please review

**Review by smarcet — APPROVED** [review]
LGTM
